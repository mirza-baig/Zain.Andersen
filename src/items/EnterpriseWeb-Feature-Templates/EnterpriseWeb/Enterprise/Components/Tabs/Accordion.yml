---
ID: "6d963802-8490-43d1-aad6-7db5c1105be5"
Parent: "d0d5ed44-d9ba-44fd-ae2c-06a65729e7c5"
Template: "0437fee2-44c9-46a6-abe9-28858d9fee8c"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/Tabs/Accordion
SharedFields:
- ID: "ba3f86a2-4a1c-4d78-b63d-91c2779c1b5e"
  Hint: __Sortorder
  Value: 50
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20230217T160455Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "1c00ce9e-6a0e-46b7-bf1a-0ef375a5ccee"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20230217T161704Z
