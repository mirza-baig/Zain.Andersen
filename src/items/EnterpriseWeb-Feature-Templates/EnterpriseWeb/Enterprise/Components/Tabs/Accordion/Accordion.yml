---
ID: "4dce5bc6-9e56-4365-890a-9a9fe6ddb5c0"
Parent: "6d963802-8490-43d1-aad6-7db5c1105be5"
Template: "ab86861a-6030-46c5-b394-e8f99e8b87db"
Path: /sitecore/templates/Feature/EnterpriseWeb/Enterprise/Components/Tabs/Accordion/Accordion
SharedFields:
- ID: "06d5295c-ed2f-4a54-9bf2-26228d113318"
  Hint: __Icon
  Value: /~/icon/office/32x32/elements_tree.png
- ID: "12c33f3f-86c5-43a5-aeb4-5598cec45116"
  Hint: __Base template
  Value: |
    {B61A1D20-9EE5-4CCB-9E0D-24A97E58D293}
    {99969A89-B62F-4155-A052-0F54C8C21B52}
    {5E2F3CF0-45EC-4618-AA30-0465290A248D}
- ID: "f7d48a55-2158-4f02-9356-756654404f73"
  Hint: __Standard values
  Value: "{5312D32D-34F2-49DD-842D-E22C243445E5}"
Languages:
- Language: en
  Versions:
  - Version: 1
    Fields:
    - ID: "25bed78c-4957-4165-998a-ca1b52f67497"
      Hint: __Created
      Value: 20230217T160511Z
    - ID: "52807595-0f8f-4b20-8d2a-cb71d28c6103"
      Hint: __Owner
      Value: |
        sitecore\Admin
    - ID: "5dd74568-4d4b-44c1-b513-0af5f4cda34f"
      Hint: __Created by
      Value: |
        sitecore\Admin
    - ID: "8cdc337e-a112-42fb-bbb4-4143751e123f"
      Hint: __Revision
      Value: "41ba9f78-0453-4a1e-8efb-d6dc88c161d6"
    - ID: "badd9cf9-53e0-4d0c-bcc0-2d784c282f6a"
      Hint: __Updated by
      Value: |
        sitecore\Admin
    - ID: "d9cf14b1-fa16-4ba6-9288-e8a174d4d522"
      Hint: __Updated
      Value: 20240103T180327Z
