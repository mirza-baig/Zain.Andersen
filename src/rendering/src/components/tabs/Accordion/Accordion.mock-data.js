const defaultData = {
    rendering: {
        dataSource: 'sampledatasource',
        componentName: 'Accordion',
    },
    "fields": {
        "componentSpacing": null,
        "sectionId": {
          "value": ""
        },
        "headlineText": {
          "value": "Accordion Headline"
        },
        "headlineLevel": {
          "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
          "url": "https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
          "name": "Heading 2",
          "displayName": "Heading 2",
          "fields": {
            "Value": {
              "value": "h2"
            }
          },
          "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
          "templateName": "Enum"
        },
        "eventName": {
          "value": ""
        },
        "eventType": {
          "value": ""
        },
        "eventZone": {
          "value": ""
        },
        "children": [
         {
    "id": "1274e64f-7a80-40de-9a3f-61dfa09a45f2",
    "displayName": "Accordion Section 2",
    "name": "Accordion Section 2",
    "templateId": "81d84b05-7e23-4528-81af-2031ef679550",
    "templateName": "Accordion Section",
    "url": "/data/components/tabs/accordion/accordion-section-2",
    "fields": {
        "sectionAnchorName": {
            "value": "section1"
        },
        "sectionTitle": {
            "value": "Accordion Section 1"
        },
        "children": []
    },
    "placeholders": {
        "components": [
            {
                "componentName": "PromoGeneric",
                "uid": "684b185b-1a9c-474c-a0e7-97e016334b93",
                "dataSource": "{684B185B-1A9C-474C-A0E7-97E016334B93}",
                "fields": {
                    "backgroundColor": {
                        "id": "e862db8d-6e1a-44a5-b255-1df40109dfa1",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/background-color-promo-generic/white",
                        "name": "White",
                        "displayName": "White",
                        "fields": {
                            "Value": {
                                "value": "white"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "bottomCaptionDescription": {
                        "value": "<p>Bottom description goes here and spans to two line text over here dummy text goes here</p>"
                    },
                    "bottomCaptionHeadline": {
                        "value": "Bottom top copy goes here and spans to two line text over here dummy text goes here"
                    },
                    "imageRatio": null,
                    "priceLevel": null,
                    "useLegalCopyFont": {
                        "value": false
                    },
                    "componentSpacing": null,
                    "sectionId": {
                        "value": ""
                    },
                    "imgPosition": {
                        "id": "870a817e-d7b9-41c7-a4c3-5c0008f2b2c1",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/image-position/left",
                        "name": "Left",
                        "displayName": "Left",
                        "fields": {
                            "Value": {
                                "value": "left"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "headlineLevel": {
                        "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-2",
                        "name": "Heading 2",
                        "displayName": "Heading 2",
                        "fields": {
                            "Value": {
                                "value": "h2"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "headlineText": {
                        "value": "Headline with right image 1:1 goes here"
                    },
                    "eyebrowLevel": null,
                    "eyebrowText": {
                        "value": ""
                    },
                    "body": {
                        "value": "Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque. Delectus et veritatis voluptas quia ut. Mollitia dignissimos doloribus odit sit et est. <a href=\"https://www.invisionapp.com/inside-design/aspect-ratio/\">Beatae</a> quae sit sit <a href=\"/\">consectetur</a>."
                    },
                    "cta1AriaLabel": {
                        "value": ""
                    },
                    "cta1Icon": {
                        "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/cta-icons/arrow",
                        "name": "Arrow",
                        "displayName": "Arrow",
                        "fields": {
                            "Value": {
                                "value": "arrow"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "cta1Link": {
                        "value": {
                            "href": "/",
                            "text": "Default state",
                            "anchor": "",
                            "linktype": "internal",
                            "class": "",
                            "title": "",
                            "target": "",
                            "querystring": "",
                            "id": "{389C03C7-02A0-463C-B02B-4ED091CAAD15}"
                        }
                    },
                    "cta1Modal": null,
                    "cta1ModalLinkText": {
                        "value": ""
                    },
                    "cta1Style": {
                        "id": "49a23327-0397-4cce-a930-e76918d37c42",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/cta-styles/primary",
                        "name": "Primary",
                        "displayName": "Primary",
                        "fields": {
                            "Value": {
                                "value": "primary"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "primaryImage": {
                        "value": {
                            "src": "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Rectangle-108.png?h=395&iar=0&w=592&rev=f254b1eecbef4d9699a393b70383bf02",
                            "alt": "Image",
                            "width": "592",
                            "height": "395"
                        }
                    },
                    "primaryImageMobile": {
                        "value": {}
                    },
                    "primaryImageMobileFocusArea": null,
                    "primaryVideo": null,
                    "cta2AriaLabel": {
                        "value": ""
                    },
                    "cta2Icon": null,
                    "cta2Link": {
                        "value": {
                            "href": ""
                        }
                    },
                    "cta2Modal": null,
                    "cta2ModalLinkText": {
                        "value": ""
                    },
                    "cta2Style": null
                }
            }
        ]
    }
},{
    "id": "1274e64f-7a80-40de-9a3f-61dfa09a45f2",
    "displayName": "Accordion Section 2",
    "name": "Accordion Section 2",
    "templateId": "81d84b05-7e23-4528-81af-2031ef679550",
    "templateName": "Accordion Section",
    "url": "/data/components/tabs/accordion/accordion-section-2",
    "fields": {
        "sectionAnchorName": {
            "value": "section2"
        },
        "sectionTitle": {
            "value": "Accordion Section 2"
        },
        "children": []
    },
    "placeholders": {
        "components": [
            {
                "componentName": "PromoGeneric",
                "uid": "684b185b-1a9c-474c-a0e7-97e016334b93",
                "dataSource": "{684B185B-1A9C-474C-A0E7-97E016334B93}",
                "fields": {
                    "backgroundColor": {
                        "id": "e862db8d-6e1a-44a5-b255-1df40109dfa1",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/background-color-promo-generic/white",
                        "name": "White",
                        "displayName": "White",
                        "fields": {
                            "Value": {
                                "value": "white"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "bottomCaptionDescription": {
                        "value": "<p>Bottom description goes here and spans to two line text over here dummy text goes here</p>"
                    },
                    "bottomCaptionHeadline": {
                        "value": "Bottom top copy goes here and spans to two line text over here dummy text goes here"
                    },
                    "imageRatio": null,
                    "priceLevel": null,
                    "useLegalCopyFont": {
                        "value": false
                    },
                    "componentSpacing": null,
                    "sectionId": {
                        "value": ""
                    },
                    "imgPosition": {
                        "id": "870a817e-d7b9-41c7-a4c3-5c0008f2b2c1",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/image-position/left",
                        "name": "Left",
                        "displayName": "Left",
                        "fields": {
                            "Value": {
                                "value": "left"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "headlineLevel": {
                        "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/headline-levels/heading-2",
                        "name": "Heading 2",
                        "displayName": "Heading 2",
                        "fields": {
                            "Value": {
                                "value": "h2"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "headlineText": {
                        "value": "Headline with right image 1:1 goes here"
                    },
                    "eyebrowLevel": null,
                    "eyebrowText": {
                        "value": ""
                    },
                    "body": {
                        "value": "Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque. Delectus et veritatis voluptas quia ut. Mollitia dignissimos doloribus odit sit et est. <a href=\"https://www.invisionapp.com/inside-design/aspect-ratio/\">Beatae</a> quae sit sit <a href=\"/\">consectetur</a>."
                    },
                    "cta1AriaLabel": {
                        "value": ""
                    },
                    "cta1Icon": {
                        "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/cta-icons/arrow",
                        "name": "Arrow",
                        "displayName": "Arrow",
                        "fields": {
                            "Value": {
                                "value": "arrow"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "cta1Link": {
                        "value": {
                            "href": "/",
                            "text": "Default state",
                            "anchor": "",
                            "linktype": "internal",
                            "class": "",
                            "title": "",
                            "target": "",
                            "querystring": "",
                            "id": "{389C03C7-02A0-463C-B02B-4ED091CAAD15}"
                        }
                    },
                    "cta1Modal": null,
                    "cta1ModalLinkText": {
                        "value": ""
                    },
                    "cta1Style": {
                        "id": "49a23327-0397-4cce-a930-e76918d37c42",
                        "url": "/sitecore/content/andersencorporation/enterprise-global/enums/cta-styles/primary",
                        "name": "Primary",
                        "displayName": "Primary",
                        "fields": {
                            "Value": {
                                "value": "primary"
                            }
                        },
                        "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
                        "templateName": "Enum"
                    },
                    "primaryImage": {
                        "value": {
                            "src": "https://edge.sitecorecloud.io/andersencorporation-m9n8s20s/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Rectangle-108.png?h=395&iar=0&w=592&rev=f254b1eecbef4d9699a393b70383bf02",
                            "alt": "Image",
                            "width": "592",
                            "height": "395"
                        }
                    },
                    "primaryImageMobile": {
                        "value": {}
                    },
                    "primaryImageMobileFocusArea": null,
                    "primaryVideo": null,
                    "cta2AriaLabel": {
                        "value": ""
                    },
                    "cta2Icon": null,
                    "cta2Link": {
                        "value": {
                            "href": ""
                        }
                    },
                    "cta2Modal": null,
                    "cta2ModalLinkText": {
                        "value": ""
                    },
                    "cta2Style": null
                }
            }
        ]
    }
}
        ]
      }
};

export default defaultData;
