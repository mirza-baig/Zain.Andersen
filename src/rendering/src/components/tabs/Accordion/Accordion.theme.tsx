// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const AccordionTheme: ThemeFile = {
  aw: {
    classes: {
      headline: 'px-m md:max-w-screen-lg md:mx-auto text-sm-m md:text-m font-heavy mb-s ',
      bodyCopy:
        'mx-auto mb-s flex w-[calc(100%_-_3rem)] items-center  justify-between pb-1 text-theme-body md:max-w-screen-lg   lg:w-full lg:px-m',
      sectionTitleWrapper:
        'sectionTitleWrapper border-t  py-m border-gray md:max-w-screen-lg  w-[calc(100%_-_3rem)] mx-auto md:px-m flex lg:w-full items-center justify-between hover:underline hover:underline-offset-4   group',
      sectiontitle: ' text-sm-s md:text-s font-heavy max-w-[80%]',
      sectionIcon:
        'border-2 border-primary rounded-full h-ml w-ml flex items-center justify-center group-hover:bg-primary  group-hover:text-white',
      accordionSection:
        '[&:last-child_.accordion-content]:after:w-[calc(100%_-_3rem)] lg:[&:last-child_.accordion-content]:after:w-full [&:last-child_.accordion-content]:after:border-t [&:last-child_.accordion-content]:after:border-gray  [&:last-child_.accordion-content]:after:max-w-screen-lg [&:last-child_.accordion-content]:after:mx-auto [&:last-child_.accordion-content]:after:block',
    },
  },
  rba: {
    classes: {
      headline:
        'px-m md:max-w-screen-lg md:mx-auto text-sm-m md:text-l font-extra-light mb-xxs md:mb-s',
      bodyCopy:
        'mx-auto mb-s flex w-[calc(100%_-_3rem)] items-center  justify-between pb-1 text-theme-body md:max-w-screen-lg   lg:w-full lg:px-m',
      sectionTitleWrapper:
        'sectionTitleWrapper border-t  py-s md:max-w-screen-lg  px-xxs w-[calc(100%_-_3rem)] mx-auto  flex lg:w-full items-center justify-between hover:underline hover:underline-offset-4 md:px-m group',
      sectiontitle:
        'text-sm-xs md:text-s font-bold md:font-medium max-w-[80%] !font-serif md:!font-sans',
      sectionIcon: 'group-hover:text-primary',
      accordionSection:
        '[&:last-child_.accordion-content]:after:w-[calc(100%_-_3rem)] lg:[&:last-child_.accordion-content]:after:w-full [&:last-child_.accordion-content]:after:border-t  [&:last-child_.accordion-content]:after:max-w-screen-lg [&:last-child_.accordion-content]:after:mx-auto [&:last-child_.accordion-content]:after:block',
    },
  },
};
