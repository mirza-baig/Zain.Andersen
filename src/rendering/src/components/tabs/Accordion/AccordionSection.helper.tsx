import { Placeholder, RouteData } from '@sitecore-jss/sitecore-jss-nextjs';

import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useEffect, useState } from 'react';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { ThemeFile, ThemeName } from 'lib/context/ThemeContext';
import { Subheadline } from 'src/helpers/Subheadline';
import { useTheme } from 'lib/context/ThemeContext';
import { Sizes } from 'src/helpers/SvgIcon/SvgIcon';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import classNames from 'classnames';
type accordionSectionProps = {
  themeData: ThemeFile[ThemeName];
} & RouteData &
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.Accordion.AccordionSection;

const AccordionSection = (props: accordionSectionProps) => {
  const [isExpanded, setIsExpanded] = useState(false);

  const { themeData } = props;
  const { themeName } = useTheme();
  const { currentScreenWidth } = useCurrentScreenType();

  const sectionAnchorName = props?.fields?.sectionAnchorName?.value;

  useEffect(() => {
    const onHashChanges = () => {
      if (typeof window !== 'undefined' && window.location.hash !== '') {
        const hash = window.location.hash.replace('#', '');
        if (sectionAnchorName === hash) {
          setIsExpanded(true);
        }
      }
    };

    onHashChanges();
    window.addEventListener('hashchange', onHashChanges);
    return () => {
      window.removeEventListener('hashchange', onHashChanges);
    };
  }, [sectionAnchorName]);

  const renderSectionIcon = (isExpanded: boolean) => {
    let Iconsize: Sizes;

    themeName == 'aw'
      ? (Iconsize = 'lg')
      : (Iconsize = currentScreenWidth >= getBreakpoint('md') ? 'xxl' : 'lg');

    return isExpanded ? (
      <SvgIcon icon="minus" size={Iconsize} className={themeData.classes.sectionIcon} />
    ) : (
      <SvgIcon icon="plus" size={Iconsize} className={themeData.classes.sectionIcon} />
    );
  };

  return (
    <section
      className={classNames(
        !isExpanded
          ? '[&:last-child_.sectionTitleWrapper]:border-b'
          : themeData.classes.accordionSection
      )}
      id={props.fields?.sectionAnchorName?.value}
    >
      <div
        aria-expanded={isExpanded}
        tabIndex={0}
        role="button"
        onClick={() => setIsExpanded(!isExpanded)}
        onKeyDown={(e) => {
          if (e.key === 'Enter' || e.key === ' ') {
            setIsExpanded(!isExpanded);
          }
        }}
        className={themeData.classes.sectionTitleWrapper}
      >
        {props?.fields && (
          <Subheadline
            useTag=""
            classes={themeData.classes.sectiontitle}
            fields={{
              subheadlineText: props?.fields?.sectionTitle,
            }}
          />
        )}

        {renderSectionIcon(isExpanded)}
      </div>

      <div className="accordion-content">
        <Placeholder
          name="components"
          rendering={props}
          render={(childComponents) => {
            return childComponents.map((component, index) => (
              <div key={index} className={isExpanded ? 'block' : 'hidden'}>
                {component}
              </div>
            ));
          }}
        />
      </div>
    </section>
  );
};

export default AccordionSection;
