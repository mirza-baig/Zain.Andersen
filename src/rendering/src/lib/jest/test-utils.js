// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
