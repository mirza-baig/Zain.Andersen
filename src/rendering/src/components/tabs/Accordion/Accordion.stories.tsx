// Global
import { Story, Meta } from '@storybook/react';
// Local
import Accordion, { AccordionProps } from './Accordion';
import defaultData from './Accordion.mock-data';

export default {
  title: 'Tabs/Accordion',
  component: Accordion,
} as Meta;
const Template: Story<AccordionProps> = (props) => <Accordion {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
