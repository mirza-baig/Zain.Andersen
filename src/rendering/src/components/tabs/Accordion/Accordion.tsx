// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { AccordionTheme } from './Accordion.theme';
import { Headline } from 'src/helpers/Headline';
import AccordionSection from './AccordionSection.helper';
import { RouteData } from '@sitecore-jss/sitecore-jss/layout';
import { ComponentProps } from 'lib/types/component-props';
import { hashCode } from 'src/helpers/Component/Component';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';

export type AccordionProps =
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.Accordion.Accordion & {
    fields: {
      children: AccordionSectionProps[];
    };
  } & ComponentProps;

export type AccordionSectionProps = RouteData &
  Feature.EnterpriseWeb.Enterprise.Components.Tabs.Accordion.AccordionSection;
const Accordion = (props: AccordionProps) => {
  const { themeData } = useTheme(AccordionTheme);

  return (
    <section
      data-component="tabs/accordion"
      id={props.fields?.sectionId?.value || `id${hashCode(props.rendering?.dataSource)}`}
    >
      <div className="theme-white my-8">
        <Headline {...props} classes={themeData.classes.headline} />
        <RichTextWrapper field={props.fields?.body} classes={themeData.classes.bodyCopy} />
        {props?.fields?.children?.map((accordionSection: AccordionSectionProps, index: number) => {
          return <AccordionSection key={index} {...accordionSection} themeData={themeData} />;
        })}
      </div>
    </section>
  );
};

export default withDatasourceCheck()<AccordionProps>(Accordion);
