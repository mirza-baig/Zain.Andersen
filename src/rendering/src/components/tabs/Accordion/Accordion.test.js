// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import Accordion from './Accordion';
import defaultData from './Accordion.mock-data';

it('renders correctly', async () => {
  const component = snapshot(Accordion, { componentProps: defaultData });
  await hasDataComponent(component, 'tabs/accordion');
});
