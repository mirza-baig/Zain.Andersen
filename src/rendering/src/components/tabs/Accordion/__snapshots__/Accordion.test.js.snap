// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="tabs/accordion"
  id="id1695242607"
>
  <div
    class="theme-white my-8"
  >
    <div
      class="items-top font-sans [&_a:hover]:underline px-m md:max-w-screen-lg md:mx-auto text-sm-m md:text-m font-heavy mb-s "
    >
      <h2>
        Accordion Headline
      </h2>
    </div>
    <section
      class="[&:last-child_.sectionTitleWrapper]:border-b"
      id="section1"
    >
      <div
        aria-expanded="false"
        class="sectionTitleWrapper border-t  py-m border-gray md:max-w-screen-lg  w-[calc(100%_-_3rem)] mx-auto md:px-m flex lg:w-full items-center justify-between hover:underline hover:underline-offset-4   group"
        role="button"
        tabindex="0"
      >
        <div
          class="font-sans  text-sm-s md:text-s font-heavy max-w-[80%]"
        >
          Accordion Section 1
        </div>
        <span
          class="border-2 border-primary rounded-full h-ml w-ml flex items-center justify-center group-hover:bg-primary  group-hover:text-white"
        />
      </div>
      <div
        class="accordion-content"
      >
        <div
          class="hidden"
        >
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              PromoGeneric
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="[&:last-child_.sectionTitleWrapper]:border-b"
      id="section2"
    >
      <div
        aria-expanded="false"
        class="sectionTitleWrapper border-t  py-m border-gray md:max-w-screen-lg  w-[calc(100%_-_3rem)] mx-auto md:px-m flex lg:w-full items-center justify-between hover:underline hover:underline-offset-4   group"
        role="button"
        tabindex="0"
      >
        <div
          class="font-sans  text-sm-s md:text-s font-heavy max-w-[80%]"
        >
          Accordion Section 2
        </div>
        <span
          class="border-2 border-primary rounded-full h-ml w-ml flex items-center justify-center group-hover:bg-primary  group-hover:text-white"
        />
      </div>
      <div
        class="accordion-content"
      >
        <div
          class="hidden"
        >
          <div
            style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;"
          >
            <h2>
              PromoGeneric
            </h2>
            <p>
              JSS component is missing React implementation. See the developer console for more information.
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>
</section>
`;
